src/etc/inc/services.inc

```diff
@@ -529,7 +613,11 @@ function services_dhcpdv4_configure() {
 						} else {
 							$itemtype = "text";
 						}
-						$custoptions .= "option custom-{$dhcpif}-{$poolidx}-{$itemidx} code {$item['number']} = {$itemtype};\n";
+                                                $optionname = "custom-{$dhcpif}-{$poolidx}-{$itemidx}";
+                                                if (array_key_exists($item['type'], $dhcpoptionsname)) {
+                                                	$optionname = $dhcpoptionsname[$item['type']];
+                                                }
+						$custoptions .= "option $optionname code {$item['number']} = {$itemtype};\n";
```

Comment:
Thank you so much for reviewing and pointing me in the right direction.   Any preferences or direction about PHP implementation is very welcome.

I'd like to also look at letting the user choose options by name, as you suggested, but if it's OK I'll start by trying to correct the proposed code.  Again, thanks!